core/src/main/scala-3/neotypes/CypherStringInterpolator.scala

```diff
@@ -0,0 +1,99 @@
+package neotypes
+package query
+
+import scala.quoted.*
+import internal.utils.createQuery
+import neotypes.model.query.QueryParam
+
+final case class CypherStringInterpolator(private val sc: StringContext) extends AnyVal {
+  inline def c(inline args: Any*): DeferredQueryBuilder = ${ CypherStringInterpolator.macroImpl('args, 'sc) }
+}
+
+object CypherStringInterpolator {
+
+  type Res = Either[String, QueryArg]
+
+  def macroImpl(argsExpr: Expr[Seq[Any]], clz: Expr[StringContext])(using quotes: Quotes): Expr[DeferredQueryBuilder] =
+    import quotes.reflect.*
+
+    def invert[T: Type](exprs: Expr[Seq[T]]): List[Expr[T]] =
+      exprs match {
+        case Varargs(props)                => props.toList
+        case '{ Seq(${ Varargs(props) }) } => props.toList.map(_.asExprOf[T])
+        case _                             => println(s"Unable to match ${exprs.show} "); null
```

Comment:
I think there is a better way to fail a macro with a compilation error.

Reply:
Yeah, thanks for reminding me of `report.errorAndAbort`.

Reply:
Addressed.